src/network/tarcap/new_taraxa_capability.cpp

```diff
@@ -121,26 +97,37 @@ TaraxaCapability::TaraxaCapability(std::weak_ptr<dev::p2p::Host> _host, NetworkC
   // Creates periodic events
   const auto lambda_ms_min = pbft_mgr ? pbft_mgr->getPbftInitialLambda() : 2000;
 
-  // TODO: use sendTransactions() from TransactionPacketHandler when ready
-  //  if (conf.network_transaction_interval > 0) {
-  //    periodic_events_tp_.post_loop( { conf.network_transaction_interval }, [this] { sendTransactions(); });
-  //  }
-
-  const auto &handler = packets_handlers_->getSpecificHandler(SubprotocolPacketType::StatusPacket);
-  auto status_packet_handler = std::static_pointer_cast<StatusPacketHandler>(handler);
+  // Send new txs periodic event
+  const auto &tx_handler = packets_handlers_->getSpecificHandler(SubprotocolPacketType::TransactionPacket);
+  auto tx_packet_handler = std::static_pointer_cast<TransactionPacketHandler>(tx_handler);
+  if (conf.network_transaction_interval > 0) {
+    periodic_events_tp_.post_loop(
+        {conf.network_transaction_interval}, [tx_packet_handler = std::move(tx_packet_handler), trx_mgr] {
+          tx_packet_handler->onNewTransactions(trx_mgr->getNewVerifiedTrxSnapShotSerialized(), false);
+        });
+  }
 
-  const auto check_alive_interval = 6 * lambda_ms_min;
-  periodic_events_tp_.post_loop({check_alive_interval},
-                                [this, status_packet_handler] { status_packet_handler->checkLiveness(); });
+  // Check liveness periodic event
+  const auto &status_handler = packets_handlers_->getSpecificHandler(SubprotocolPacketType::StatusPacket);
+  auto status_packet_handler = std::static_pointer_cast<StatusPacketHandler>(status_handler);
+  const auto check_liveness_interval = 6 * lambda_ms_min;
+  periodic_events_tp_.post_loop({check_liveness_interval}, [status_packet_handler = std::move(status_packet_handler)] {
+    status_packet_handler->checkLiveness();
+  });
 
+  // Logs packets stats periodic event
   if (conf.network_performance_log_interval > 0) {
     periodic_events_tp_.post_loop({conf.network_performance_log_interval},
-                                  [this] { peers_state_->packets_stats_.logStats(); });
+                                  [peers_state = peers_state_] { peers_state->packets_stats_.logStats(); });
   }
 
-  // TODO: implement logNodeStats()
-  //  const auto summary_interval_ms = 5 * 6 * lambda_ms_min;
-  //  periodic_events_tp_.post_loop( { summary_interval_ms }, [this] { logNodeStats(); });
+  // Logs node stats periodic event
+  const auto node_stats_log_interval = 5 * 6 * lambda_ms_min;
+  // node_stats is shared_ptr as it is probably less expensive to move periodically than object with uint64_t members
+  auto node_stats = std::make_shared<NodeStats>(peers_state_, syncing_state_, pbft_chain, pbft_mgr, dag_mgr,
+                                                dag_blk_mgr, vote_mgr, trx_mgr, node_stats_log_interval);
+  periodic_events_tp_.post_loop({node_stats->getNodeStatsLogInterval()},
+                                [node_stats = std::move(node_stats)]() mutable { node_stats->logNodeStats(); });
```

Comment:
does it need to be mutable when using shr_ptr? I do know just asking

Reply:
fixed, no shared_ptr used anymore